### ISSUE TITLE
Unable to put several items in an interation

### ISSUE THREAD

**Issue by jotadrilo** [devicescape/aws_dynamo#5]
I'm trying to iterate generating some put-item request, the method is:

1 - Create a string request (nothing wrong here)
2 - Initialize aws (aws_init)
3 - Put Item (aws_dynamo_put_item)
4 - Deinitialize aws (aws_deinit)
5 - Free response (aws_dynamo_free_put_item_response)
6 - Go 1 again 

The iteration end when all the items are processed, but only the first one is ok.
And I get this message for other case:
   WARN:  aws_post: HTTP post failed, will retry.
   WARN:  aws_post: Retry failed.
   WARN:  aws_dynamo_request: Post failed.

**Comment by jotadrilo** [comment]
I solved this putting all the items before the deinitialization. This may not be like this in some cases.

**Comment by dkimdon** [comment]
aws_init should be called at application startup, and aws_deinit should be
called at application shutdown.  You don't want to be calling those two
inside a loop the way it sounds like you are doing it.

For example, if you get credentials from an IAM role then the credentials
are obtained multiple times if you keep init'ing and deinit'ing the
library.  That is a waste of resources, your application will be much
slower.

The handle returned by aws_init shoudl be used as a long-lived handle, use
it for many, many DynamoDB requests over a period of months or years.

**Comment by jotadrilo** [comment]
I tried that but the handle didnt work in another context than the initialization context.

I mean, i was getting a pointer to the handle. I had a structure reachable by other functions (passing it) and this pointer was defined inside. 

Debugging and watching the pointer I saw that it was correctly formed. However I was not able to use it with the dynamo put item function. I got the error message that i said in the first post.

**Comment by dkimdon** [comment]
Pass  --enable-debug  to configure and you will see more details of what
DynamoDB is returning (or turn off ssl and use a network sniffer like
wireshark).  That will help us to understand why you are unable to reuse
the handle.

**Comment by jotadrilo** [comment]
This has been really hard to solve:

I created a empty new project (lets call it B) which do only the init-put-deinit process: Everything was fine.
So I debugged both projects since the put_item call and I watched the ssl handshake with Wireshark.

While the process B sent the Client Hello, the process A (to fix) didn't.
So the problem had to be in the curl perform.

I added this instruction to the line 500 of http.c: curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
I recompiled and I launched both projects again.
I got some like this from curl: SSL_CTX_new:unable to load ssl2 md5 routines

That was the problem. I used openssl functions previously and I cleaned up the context. Curl require this context and it is unable to reload them because don't count the clean up calls to figure out.

So I commented the clean-up line -EVP_cleanup()- and it works fine now.
